### PR TITLE
cardano-node: 1.20.0 -> 1.21.1

### PR DESCRIPTION
**Pull request by disassembler**
### Issue Number

None

### Overview

Update for [cardano-node v1.21.1](https://github.com/input-output-hk/cardano-node/releases/tag/1.21.1).

- [x] bumps niv pin for cardano-node.
- [x] Merge stack snapshot yaml for cardano 1.21.1 (PR input-output-hk/cardano-haskell#30).
- [x] Update cardano snapshot version in stack.yaml.
- [x] Fix compile errors.
- [ ] ~Try to import symbols from `Cardano.Api.*` instead of `Shelley.Spec.Ledger.*`, etc, where possible (see diff for new exports).~ (I attempted this and it's rather painful - better to do it later)

### Comments

- [cardano-node diff 1.20.0..1.21.1](https://github.com/input-output-hk/cardano-node/compare/1.20.0...1.21.1)

### CODE REVIEW

**Review by Anviking — APPROVED** [review]
lgtm

**Review by hasufell — DISMISSED** [review]
Cabal config files need to be regenerated, see

https://github.com/input-output-hk/cardano-wallet/wiki/Building#syncing-stack-and-cabal-dependencies

Also related: https://github.com/input-output-hk/cardano-wallet/pull/2204

**Comment by rvl** [comment]
bors try

**Comment by hasufell** [comment]
> You do it, please.

Sure, but it's something that everyone should be able to do, just like regenerating the nix expressions.

**Comment by hasufell** [comment]
bors try

File: stack.yaml [hunk 1]
```diff
@@ -79,3 +79,185 @@ nix:
   # Disabling the pure nix-shell allows environment variables to be
   # passed down to tests. We need this for integration tests.
   pure: false
+
+
+# !!! This can be dropped when cabal-3.4.0.0 is released !!!
+#
+#+BEGIN_STACK2CABAL
```
Comment (KtorZ) [inline stack.yaml:86]: I still don't think that this is the right place for this. But let's see.

**Review by KtorZ — APPROVED** [review]

**Comment by hasufell** [comment]
https://github.com/input-output-hk/cardano-wallet/issues/2196 seems related

**Pull request merged**